nixos/modules/virtualisation/nixos-containers.nix

```diff
@@ -4,6 +4,10 @@ with lib;
 
 let
 
+configurationPrefix = optionalString (versionAtLeast config.system.stateVersion "20.09") "nixos-";
+configurationDirectory = "/etc/${configurationPrefix}containers";
+stateDirectory = "/var/lib/${configurationPrefix}containers";
```

Comment:
Hmm, but since we're not supposed to bump the state-version on a system-update, wouldn't this break 20.03 installations when upgrading to 20.09 if there are existing nixos-containers? To fix this, I'd suggest to raise a warning if `/var/lib/containers` is used.

Reply:
When you upgrade, you generally don't change `stateVersion`. So existing installations that upgrade to 20.09 would continue to use `/var/lib/containers` and they wouldn't break (unless you use podman or other tools).

Reply:
I'm well-aware of that, but if you deploy a podman instance to an existing system you'll run into issues (as you already mentioned), that's why I brought this up :)